From c20654f0d561225f21ec68428ccfff8e30223881 Mon Sep 17 00:00:00 2001
From: Lu Hui <luhui@sipeed.com>
Date: Fri, 29 Dec 2023 11:58:32 +0800
Subject: [PATCH] lib/c-stack.c: fix build on ubuntu 22.04.3

---
 lib/c-stack.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/c-stack.c b/lib/c-stack.c
index 9bbe6fe..9ccbe7a 100644
--- a/lib/c-stack.c
+++ b/lib/c-stack.c
@@ -52,12 +52,12 @@ typedef struct sigaltstack stack_t;
 #endif
 #ifndef SIGSTKSZ
 # define SIGSTKSZ 16384
-#elif HAVE_LIBSIGSEGV && SIGSTKSZ < 16384
+//#elif HAVE_LIBSIGSEGV && SIGSTKSZ < 16384
 /* libsigsegv 2.6 through 2.8 have a bug where some architectures use
    more than the Linux default of an 8k alternate stack when deciding
    if a fault was caused by stack overflow.  */
-# undef SIGSTKSZ
-# define SIGSTKSZ 16384
+//# undef SIGSTKSZ
+//# define SIGSTKSZ 16384
 #endif
 
 #include <stdlib.h>
-- 
2.41.0

