diff --git a/system/init.d/S95nanokvm b/system/init.d/S95nanokvm
index b028421..b480a52 100755
--- a/kvmapp/system/init.d/S95nanokvm
+++ b/kvmapp/system/init.d/S95nanokvm
@@ -21,8 +21,8 @@ case "$1" in
     cp -r /kvmapp/kvm_system /tmp/
     /tmp/kvm_system/kvm_system &
     
-    # cp -r /kvmapp/server /tmp/
-    # /tmp/server/NanoKVM-Server &
+    cp -r /kvmapp/server /tmp/
+    /tmp/server/NanoKVM-Server &
   ;;
   stop)
     killall kvm_system
@@ -40,8 +40,8 @@ case "$1" in
     cp -r /kvmapp/kvm_system /tmp/
     /tmp/kvm_system/kvm_system &
 
-    # cp -r /kvmapp/server /tmp/
-    # /tmp/server/NanoKVM-Server &
+    cp -r /kvmapp/server /tmp/
+    /tmp/server/NanoKVM-Server &
 
     sync
 
