From 482d93cbf4ac61a420ee41f25acc4c32745459c3 Mon Sep 17 00:00:00 2001
From: scpcom <scpcom@gmx.de>
Date: Tue, 29 Oct 2024 01:42:41 +0100
Subject: [PATCH] genconfig: disable menuconfig

---
 tools/kconfig/genconfig.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/kconfig/genconfig.py b/tools/kconfig/genconfig.py
index 8d58f7b..c787cd7 100644
--- a/tools/kconfig/genconfig.py
+++ b/tools/kconfig/genconfig.py
@@ -10,7 +10,7 @@ kconfig_lib_path = sys.path[0]+"/Kconfiglib"
 sys.path.append(kconfig_lib_path)
 
 import kconfiglib
-from menuconfig import menuconfig
+#from menuconfig import menuconfig
 
 
 def _cmake_contents(kconfig, header):
@@ -153,7 +153,7 @@ else:
     kconfig.load_config()
 
 if args.menuconfig == "True":
-    menuconfig(kconfig)
+    print("menuconfig(kconfig) disabled!")
 
 # write back
 for fmt, filename in out_format.items():
-- 
2.34.1

