From 9b1c688aba35920e15e7871a4b90e8cba80112d9 Mon Sep 17 00:00:00 2001
From: scpcom <scpcom@gmx.de>
Date: Tue, 10 Dec 2024 18:05:30 +0100
Subject: [PATCH] server: isLibExist: change path to /kvmapp/server/dl_lib

---
 server/service/auth/login.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/service/auth/login.go b/server/service/auth/login.go
index 29da0d5..44cea41 100644
--- a/server/service/auth/login.go
+++ b/server/service/auth/login.go
@@ -66,7 +66,7 @@ func (s *Service) Login(c *gin.Context) {
 }
 
 func isLibExist() bool {
-	libPath := fmt.Sprintf("/kvmapp/kvm_system/dl_lib/libmaixcam_lib.so")
+	libPath := fmt.Sprintf("/kvmapp/server/dl_lib/libmaixcam_lib.so")
 	_, err := os.Stat(libPath)
 	return err == nil
 }
-- 
2.34.1

