From ef41f8c595b3dcffebf84756fa2aecfff54e35ce Mon Sep 17 00:00:00 2001
From: scpcom <scpcom@gmx.de>
Date: Sun, 23 Mar 2025 02:21:39 +0100
Subject: [PATCH] server: default disable stun

---
 server/config/default.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/server/config/default.go b/server/config/default.go
index a841e9e..9f777a5 100644
--- a/server/config/default.go
+++ b/server/config/default.go
@@ -19,7 +19,7 @@ var defaultConfig = &Config{
 		RefreshTokenDuration: 2678400,
 		RevokeTokensOnLogout: true,
 	},
-	Stun: "stun.l.google.com:19302",
+	Stun: "disable",
 	Turn: Turn{
 		TurnAddr: "",
 		TurnUser: "",
@@ -39,7 +39,7 @@ func checkDefaultValue() {
 	}
 
 	if instance.Stun == "" {
-		instance.Stun = "stun.l.google.com:19302"
+		instance.Stun = "disable"
 	}
 
 	if instance.Authentication == "" {
-- 
2.34.1

